### DISCUSSION TITLE
How to solve the problem that fsevents.node in chokidar is compiled during the compilation phase of webpack production packaging?

### DISCUSSION THREAD

**Discussion by white-than-wood** [webpack/webpack#15851] [General]
## Feature request

have a way to ignore require calls!

**What is the expected behavior?**

<img width="818" alt="image" src="https://user-images.githubusercontent.com/11425476/169497726-cfc8881c-af22-41f7-8082-498bc08dc5b5.png">

**What is motivation or use case for adding/changing the behavior?**

The thing is this, I use electron-reload in the main process typescript file of electron to do hot loading, electron-reload depends on chokidar (fsevents.node) and needs to use the node-loader module to compile, but I only want to use it in the development environment used in , so I wrote it like this:

<img width="822" alt="image" src="https://user-images.githubusercontent.com/11425476/169498395-0678c95f-00c8-456e-8cf9-e8b7701b564e.png">

And also set the node-loader to be used in the development environment.

However, due to its typescript static ES module compilation, chokidar (fsevents.node) is still introduced at compile time. At this time, there is no node-loader module for compilation. Is there any way I can do it without using node-loader? Next, ignore electron-reload, I need this function urgently.

![image](https://user-images.githubusercontent.com/11425476/169498324-b9478afa-4c6b-4c23-8027-d45082b24027.png)
                                                    When node-loader is not used in production environment.

**How should this be implemented in your opinion?**

Perhaps,like webpack-node-externals, I can specify which modules you want to ignore

**Are you willing to work on this yourself?**
yes

**Comment by alexander-akait** [comment]
You can use https://webpack.js.org/plugins/ignore-plugin/ or externals with empty module

**Comment by white-than-wood** [comment]
Isn't ignoreplugin ignoring some modules in the dependencies that are not used? Can I ignore all modules?and Is this way of externals too rude, and are there any side effects?

**Reply by alexander-akait** [reply]
You can use `contextRegExp` to exclude file when you need it and ignore modules which you need, anyway you should avoid Node.js modules for browser, it is strong recomendation or ask developer to provide support for browser, but it is already solved https://github.com/chalk/chalk/blob/main/package.json#L15, just use browser version

**Reply by white-than-wood** [reply]
> You can use `contextRegExp` to exclude file when you need it and ignore modules which you need, anyway you should avoid Node.js modules for browser, it is strong recomendation or ask developer to provide support for browser, but it is already solved https://github.com/chalk/chalk/blob/main/package.json#L15, just use browser version

ok,I'll try it!
